src/Appwrite/Platform/Tasks/Hamster.php

```diff
@@ -378,96 +151,43 @@ protected function calculateByGroup(string $collection, Database $dbForConsole,
-                    ->setProps($statsPerOrganization);
-                $res = $this->mixpanel->createEvent($event);
-                if (!$res) {
-                    throw new Exception('Failed to create event for organization : ' . $document->getId());
-                }
+                $queue = $pools->get('queue')->pop();
+                $connection = $queue->getResource();
+    
+                $hamsterTask = new EventHamster($connection);
+    
+                $hamsterTask
+                    ->setType('organization')
+                    ->setOrganization($organization)
+                    ->trigger();
+    
+                $queue->reclaim();
             } catch (Exception $e) {
                 Console::error($e->getMessage());
             }
         });
     }
 
-    protected function getStatsPerUser(Database $dbForConsole)
+    protected function getStatsPerUser(Group $pools, Database $dbForConsole)
     {
-        $this->calculateByGroup('users', $dbForConsole, function (Database $dbForConsole, Document $document) {
+        $this->calculateByGroup('users', $dbForConsole, function (Database $dbForConsole, Document $user) use ($pools) {
             try {
-                $statsPerUser = [];
-
-                /** Organization name */
-                $statsPerUser['name'] = $document->getAttribute('name');
-
-                /** Organization ID (needs to be stored as an email since mixpanel uses the email attribute as a distinctID) */
-                $statsPerUser['email'] = $document->getAttribute('email');
-
-                /** Organization Creation Date */
-                $statsPerUser['created'] = $document->getAttribute('$createdAt');
-
-                /** Number of teams this user is a part of */
-                $statsPerUser['memberships'] = $dbForConsole->count('memberships', [
-                    Query::equal('userInternalId', [$document->getInternalId()]),
-                    Query::limit(APP_LIMIT_COUNT)
-                ]);
-
-                if (!isset($statsPerUser['email'])) {
-                    throw new Exception('User has no email: ' . $document->getId());
-                }
-
-                /** Send data to mixpanel */
-                $event = new Event();
-                $event
-                    ->setName('User Daily Usage')
-                    ->setProps($statsPerUser);
-                $res = $this->mixpanel->createEvent($event);
-
-                if (!$res) {
-                    throw new Exception('Failed to create user profile for user: ' . $document->getId());
-                }
+                $queue = $pools->get('queue')->pop();
+                $connection = $queue->getResource();
+    
+                $hamsterTask = new EventHamster($connection);
+    
+                $hamsterTask
+                    ->setType('user')
+                    ->setUser($user)
+                    ->trigger();
+    
+                $queue->reclaim();
```

Comment:
We dont need to reclaim the queue after each iteration.. This can be done in the end after the entire loop is complete